apache/zeppelin
# Using HDFS to backup and restore notebook

### What is this PR for?
This PR supports using HDFS to backup and restore notebook.
It is similar to https://github.com/apache/zeppelin/pull/1479.
However this PR just use WebHDFS API, so we don't need to care about hadoop libraries' dependency.

### What type of PR is it?
Improvement

### Todos

### What is the Jira issue?
[https://issues.apache.org/jira/browse/ZEPPELIN-1515](https://issues.apache.org/jira/browse/ZEPPELIN-1515)

### How should this be tested?

Set the variables in zeppelin-site.xml
- zeppelin.notebook.storage : org.apache.zeppelin.notebook.repo.HDFSNotebookRepo
- hdfs.url : Hadoop WebHDFS URL. default: http://localhost:50070/webhdfs/v1/
- hdfs.user : HDFS user. default: hdfs
- hdfs.maxlength : Maximum number of lines of results fetched. default: 100000
- hdfs.notebook.dir : notebook location directory in HDFS. default: /tmp

After zeppelin daemon start, check the notebook directories in ``hdfs.notebook.dir``.

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes
